app/controllers/coronavirus/sub_sections_controller.rb

```diff
@@ -9,11 +9,22 @@ def new
 
     def create
       @sub_section = page.sub_sections.new(sub_section_params)
-      if @sub_section.save && draft_updater.send
-        redirect_to coronavirus_page_path(page.slug), notice: "Sub-section was successfully created."
-      else
-        @sub_section.errors.add :base, draft_updater.errors.to_sentence
+      errors = nil
+
+      SubSection.transaction do
+        if @sub_section.save
+          unless draft_updater.send
+            errors = draft_updater.errors.to_sentence
+            raise ActiveRecord::Rollback
+          end
```

Comment:
I don't think the conditional is needed. I think this can be written in similar style to `update`

```
@sub_section.save!
raise ActiveRecord::Rollback unless draft_updater.send
```

Reply:
Conditional is not needed. Sorted after a talk offline 🥇